### PR TITLE
fix: don't compile right after generating a project

### PR DESCRIPTION
**Pull request by jinwoo**
Compilation would fail because there's no source files.

### CODE REVIEW

File: src/init.ts [hunk 1]
```diff
@@ -215,7 +215,9 @@ export async function init(options: Options): Promise<boolean> {
 
   // Run `npm install` after initial setup so `npm run check` works right away.
   if (!options.dryRun) {
-    cp.spawnSync('npm', ['install'], {stdio: 'inherit'});
+    // --ignore-scripts so that compilation doesn't happen because there's no
+    // source files yet.
+    cp.spawnSync('npm', ['install', '--ignore-scripts'], {stdio: 'inherit'});
```
Comment (DominicKramer) [inline src/init.ts]: If someone has a `postinstall` script, I don't think that will get invoked either.  However, I expect this use case is probably not too common and, if needed, someone can always run `npm install` manually if needed.

File: test/test-kitchen.ts [hunk 2]
```diff
@@ -113,6 +113,10 @@ test.serial('use as a non-locally installed module', async t => {
   // server.ts has a lint error. Should error.
   await t.throws(simpleExecp(`${GTS} check src/server.ts`, opts));
 
+  // Compilation shouldn't have happened. Hence no `build` directory.
+  const dirContents = fs.readdirSync(`${tmpDir.name}/kitchen`);
+  t.is(dirContents.indexOf('build'), -1);
```
Comment (DominicKramer) [inline test/test-kitchen.ts]: Is it possible that `npm run compile` was run but had an error that caused the `build` directory to not be created?

File: src/init.ts (hunk 1 above)
Comment (jinwoo) [reply src/init.ts]: Yeah, most of the cases `gts init` is run from an empty directory. Even though someone had a `postinstall` script, they would know what they are doing and be able to figure things out.

File: test/test-kitchen.ts (hunk 2 above)
Comment (jinwoo) [reply test/test-kitchen.ts]: I verified that the `build` directory is created without my change.

**Comment by jinwoo** [comment]
I moved the new test to a better place.

**Pull request merged**